keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1537 @@
+  a mechanism for cleanup of the device, that will ensure that device
+  does not contain traces/parameters/data from previous workloads and
+  appropriate power state/shutdown.
+
+  *Limitation*: Post-stop actions are not supported.
+
+- *Partial allocation*: When deploying a container I’d like to be able
+  to use part of the shareable device inside a container and other
+  containers should be able to use other free resources on the same
+  device.
+
+  *Limitation*: Current implementation of the device plugin doesn’t
+  allow one to allocate part of the device because parameters are too limited
+  and Kubernetes doesn't have enough information about the extended
+  resources on a node to decide whether they can be shared.
+
+- *Optional allocation*: When deploying a workload I’d like to specify
+  soft(optional) device requirements. If a device exists and it’s
+  allocatable it will be allocated. If not - the workload will be run on
+  a node without a device. GPU and crypto-offload engines are
+  examples of this kind of device. If they’re not available, workloads
+  can still run by falling back to using only the CPU for the same
+  task.
+
+  *Limitation*: Optional allocation is supported neither by the device
+  plugins nor by current Pod resource declaration.
+
+- *Support Over the Fabric devices*: When deploying a container, I’d
+  like to utilize devices available over the Fabric (PCIe, CXL,
+  Network, special links, etc).
+
+  *Limitation*: Device Plugins framework supports only local devices.
+
+- *Access to the container image*: When deploying a container that
+  needs access to a GPU device, I would like my container to
+  gracefully fail to start (rather than SIGSEGV at runtime) if a
+  minimum driver requirement is not met on the host.
+
+  *Limitation*: GPU driver requirements are currently stored as labels
+  on the container image, and device plugins do not have access to the
+  container image.
+
+Because this KEP enables the usage of
+[CDI](https://github.com/container-orchestrated-devices/container-device-interface/#readme)
+in Kubernetes, it also addresses those problems that are handled by
+CDI:
+
+- *Perform container runtime specific operations*: When deploying a container
+  that needs access to a device, I would like to be able to reuse the
+  same pod spec, irrespective of the underlying container runtime in
+  use (e.g. kata vs. runc).
+
+  *Limitation*: At present, there is no way to perform
+  runtime-specific operations that may need to occur as a result of
+  injecting a device into a container (device plugins are runtime
+  agnostic). A good example is supporting GPU passthrough
+  virtualization on kata vs. runc.
+
+- *Access to the plugin container*: When deploying a device plugin, I
+  would like to ensure that all of the operations that need to occur
+  as part of “injecting” a device into a container, also occur for the
+  “plugin container” itself.
```

Comment:
By "plugin container", do you mean the "third-party device plugin" that runs as a container?

Reply:
I'm not sure either what the "plugin container" here is. @kad?